core/services/chainlink/application.go

```diff
@@ -256,32 +272,78 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 				return nil, fmt.Errorf("could not configure syncer: %w", err)
 			}
 
+			workflowDonNotifier := capabilities.NewDonNotifier()
+
 			wfLauncher := capabilities.NewLauncher(
 				globalLogger,
 				externalPeerWrapper,
 				dispatcher,
 				opts.CapabilitiesRegistry,
+				workflowDonNotifier,
 			)
 			registrySyncer.AddLauncher(wfLauncher)
 
 			srvcs = append(srvcs, wfLauncher, registrySyncer)
+
+			if cfg.Capabilities().WorkflowRegistry().Address() != "" {
+				if gatewayConnectorWrapper == nil {
+					return nil, errors.New("unable to create workflow registry syncer without gateway connector")
+				}
+
+				err = keyStore.Workflow().EnsureKey(context.Background())
+				if err != nil {
+					return nil, fmt.Errorf("failed to ensure workflow key: %w", err)
+				}
+
+				keys, err := keyStore.Workflow().GetAll()
+				if err != nil {
+					return nil, fmt.Errorf("failed to get all workflow keys: %w", err)
+				}
+				if len(keys) != 1 {
+					return nil, fmt.Errorf("expected 1 key, got %d", len(keys))
+				}
+
+				connector := gatewayConnectorWrapper.GetGatewayConnector()
+				webAPILggr := globalLogger.Named("WebAPITarget")
+
+				webAPIConfig := webapi.ServiceConfig{
+					RateLimiter: common2.RateLimiterConfig{
+						GlobalRPS:      100.0,
+						GlobalBurst:    100,
+						PerSenderRPS:   100.0,
+						PerSenderBurst: 100,
+					},
+				}
+
+				outgoingConnectorHandler, err := webapi.NewOutgoingConnectorHandler(connector,
+					webAPIConfig,
+					capabilities2.MethodWebAPITarget, webAPILggr)
+				if err != nil {
+					return nil, fmt.Errorf("could not create outgoing connector handler: %w", err)
+				}
+
+				eventHandler := syncer.NewEventHandler(globalLogger, syncer.NewWorkflowRegistryDS(opts.DS, globalLogger), syncer.NewFetcherFunc(globalLogger, outgoingConnectorHandler), workflowstore.NewDBStore(opts.DS, globalLogger, clockwork.NewFakeClock()), opts.CapabilitiesRegistry,
+					custmsg.NewLabeler(), clockwork.NewRealClock(), keys[0])
+
+				loader := syncer.NewWorkflowRegistryContractLoader(cfg.Capabilities().WorkflowRegistry().Address(), func(ctx context.Context, bytes []byte) (syncer.ContractReader, error) {
+					return relayer.NewContractReader(ctx, bytes)
```

Comment:
You can simplify this to just relayer.NewContractReader I think?

Reply:
it needs to be the factory method, internally the workflow registry create this on demand